### DISCUSSION TITLE
Granularity of video recordings

### DISCUSSION THREAD

**Discussion by djMax** [wix/Detox#3277] [Q&A] [answered]
I know there has been discussion of whole-suite recording and that that's hard, BUT, I just want tests to record something useful. Right now I have tests like this: 

```
  describe('Skip All', () => {
    it('should have select country screen', async () => {
      await expect(element(by.id('ftux::choose_country'))).toBeVisible();
    });

    it('should allow changing country picker value', async () => {
      await element(by.id('ftux::country_picker')).setColumnToValue(0, '🇨🇦 Canada');
      await element(by.id('ftux::country_picker')).setColumnToValue(0, '🇺🇸 United States');
    });

    it('should show welcome screen after continue', async () => {
      await element(by.id('ftux::country_selection::cta')).tap();
      await expect(element(by.id('login_content::title'))).toBeVisible();
    });
  });
```

That will record 3 videos, which isn't what I want. But if I remove all the "its" then I will have less useful textual output. Is there a way to get recordings to work at the "describe" level instead of the "it" level?

**Comment by d4vidi** [comment, marked as answer]
@djMax we have no such capabilities ATM, I'm afraid. Nevertheless, we've had discussions over it in the past and would like to support that at _some_ point in the (far) future.

As a suggestion, you might consider introducing recording by yourself (i.e. look up the associated commands in the Detox log, and utilize yourself in `beforeAll()` and `afterAll()`).

In any case, I'm intrigued -- why are the test-granularity videos useful? I'm inclined to say that it has something to do with the fact that you do not relaunch the app or `reloadReactNative()` before each test, which isn't good practice.

**Comment by djMax** [comment]
We do reload on each "suite" - so once per file basically. But right now, our "it" tests are like 1 screen entering data and clicking "next" and then the next "it" is the next screen or next action. So the recordings end up being totally useless because they don't capture the before/after transitions, which is usually where the problem hides. I get your point about doing our own recording - that probably is the easiest path.

**Reply by d4vidi** [reply]
Well I understand the technical aspect of what your saying, but I advise to double check the best practices in terms of test isolation, as, at the moment - at least so it seems, your tests are leaking. Perhaps [this blogpost](https://medium.com/wix-engineering/detox-writing-stable-test-suites-372c9d537184) by @rotemmiz could come in handy.

**Reply by rotemmiz** [reply]
Look at me, mentioned here 🥲
